src/wallet/rpcwallet.cpp

```diff
@@ -5533,6 +5537,7 @@ UniValue z_shieldcoinbase(const UniValue& params, bool fHelp)
                             "Only Sapling shielding is currently supported by z_shieldcoinbase. "
                             "Use z_sendmany with a transaction amount that results in no change for Orchard shielding.");
                 }
+                involvesOrchard = ua.GetOrchardReceiver().has_value();
```

Comment:
Per the above error message, we currently always ignore the Orchard receiver of a UA, and thus always generate a Sapling shielding transaction. In that scenario, we should be respecting the preference flag. However, if the UA has an Orchard receiver, this is going to force a v5 transaction where it is not necessary.

@nuttycom points out that if a user has generated a UA with an Orchard receiver, their wallet almost certainly supports v5 transactions. I think this is sufficient to leave this as-is, though it's a bit ugly.

A stronger rationale is that if @nuttycom gets a PR up to add Orchard support to `z_shieldcoinbase` then this issue is moot, as the UA's preference order takes precedence.